1/(exp(it) - 2) integrates wrong

```
>>> var('t', real=True)
t
>>> (1/(exp(I*t)-2)).integrate((t, 0, 2*pi))
-ⅈ⋅π
```

Correct answer is `-pi` according to wolframalpha

Yes, that seems right:
```diff
diff --git a/sympy/polys/euclidtools.py b/sympy/polys/euclidtools.py
index c0b6d13e79..7a772a012a 100644
--- a/sympy/polys/euclidtools.py
+++ b/sympy/polys/euclidtools.py
@@ -992,8 +992,7 @@ def dup_rr_prs_gcd(f, g, K):
     h = dup_subresultants(F, G, K)[-1]
     _, h = dup_primitive(h, K)
 
-    if K.is_negative(dup_LC(h, K)):
-        c = -c
+    c *= K.canonical_unit(dup_LC(h, K))
 
     h = dup_mul_ground(h, c, K)
```
There is one test failure in polys:
```
        a, b = sqrt(-2), -sqrt(-2)
>       assert gcd(a, b) in (a, b)
E       assert sqrt(2) in (sqrt(2)*I, -sqrt(2)*I)
E        +  where sqrt(2) = gcd(sqrt(2)*I, -sqrt(2)*I)

sympy/polys/tests/test_polytools.py:2055: AssertionError
```
This is because the output has changed for this:
```python
In [1]: gcd(sqrt(-2), -sqrt(-2))
Out[1]: √2
```
On master that gives:
```python
In [1]: gcd(sqrt(-2), -sqrt(-2))
Out[1]: -√2⋅ⅈ
```
I think it's reasonable that the gcd should be canonical wrt units.

--- a/sympy/integrals/tests/test_integrals.py
+++ b/sympy/integrals/tests/test_integrals.py
@@ -1604,11 +1604,15 @@ def test_integrate_with_complex_constants():
     K = Symbol('K', real=True, positive=True)
     x = Symbol('x', real=True)
     m = Symbol('m', real=True)
+    t = Symbol('t', real=True)
     assert integrate(exp(-I*K*x**2+m*x), x) == sqrt(I)*sqrt(pi)*exp(-I*m**2
                     /(4*K))*erfi((-2*I*K*x + m)/(2*sqrt(K)*sqrt(-I)))/(2*sqrt(K))
-    assert integrate(1/(1 + I*x**2), x) == -sqrt(I)*log(x - sqrt(I))/2 +\
-        sqrt(I)*log(x + sqrt(I))/2
+    assert integrate(1/(1 + I*x**2), x) == (-I*(sqrt(-I)*log(x - I*sqrt(-I))/2
+            - sqrt(-I)*log(x + I*sqrt(-I))/2))
     assert integrate(exp(-I*x**2), x) == sqrt(pi)*erf(sqrt(I)*x)/(2*sqrt(I))
+
+    assert integrate((1/(exp(I*t)-2)), t) == -t/2 - I*log(exp(I*t) - 2)/2
+    assert integrate((1/(exp(I*t)-2)), (t, 0, 2*pi)) == -pi
 
 
 def test_issue_14241():

--- a/sympy/polys/euclidtools.py
+++ b/sympy/polys/euclidtools.py
@@ -992,8 +992,7 @@ def dup_rr_prs_gcd(f, g, K):
     h = dup_subresultants(F, G, K)[-1]
     _, h = dup_primitive(h, K)
 
-    if K.is_negative(dup_LC(h, K)):
-        c = -c
+    c *= K.canonical_unit(dup_LC(h, K))
 
     h = dup_mul_ground(h, c, K)
 

--- a/sympy/polys/tests/test_polytools.py
+++ b/sympy/polys/tests/test_polytools.py
@@ -2049,10 +2049,12 @@ def test_gcd():
     assert lcm(f, g, modulus=11, symmetric=False) == l
 
     a, b = sqrt(2), -sqrt(2)
-    assert gcd(a, b) == gcd(b, a) == a
+    assert gcd(a, b) == gcd(b, a) == sqrt(2)
 
     a, b = sqrt(-2), -sqrt(-2)
-    assert gcd(a, b) in (a, b)
+    assert gcd(a, b) == gcd(b, a) == sqrt(2)
+
+    assert gcd(Poly(x - 2, x), Poly(I*x, x)) == Poly(1, x, domain=ZZ_I)
 
     raises(TypeError, lambda: gcd(x))
     raises(TypeError, lambda: lcm(x))

